dpnp/backend/extensions/lapack/heevd.cpp

```diff
@@ -0,0 +1,210 @@
+        cgh.host_task([ctx, scratchpad]() { sycl::free(scratchpad, ctx); });
+    });
+    host_task_events.push_back(clean_up_event);
+    return heevd_event;
+}
+
+std::pair<sycl::event, sycl::event> heevd(sycl::queue exec_q,
+                                          const std::int8_t jobz,
+                                          const std::int8_t upper_lower,
+                                          dpctl::tensor::usm_ndarray eig_vecs,
+                                          dpctl::tensor::usm_ndarray eig_vals,
+                                          const std::vector<sycl::event>& depends)
+{
+    const int eig_vecs_nd = eig_vecs.get_ndim();
+    const int eig_vals_nd = eig_vals.get_ndim();
+
+    if (eig_vecs_nd != 2)
+    {
+        throw py::value_error("Unexpected ndim=" + std::to_string(eig_vecs_nd) +
+                              " of an output array with eigenvectors");
+    }
+    else if (eig_vals_nd != 1)
+    {
+        throw py::value_error("Unexpected ndim=" + std::to_string(eig_vals_nd) +
+                              " of an output array with eigenvalues");
+    }
+
+    const py::ssize_t* eig_vecs_shape = eig_vecs.get_shape_raw();
+    const py::ssize_t* eig_vals_shape = eig_vals.get_shape_raw();
+
+    if (eig_vecs_shape[0] != eig_vecs_shape[1])
+    {
+        throw py::value_error("Output array with eigenvectors with be square");
+    }
+    else if (eig_vecs_shape[0] != eig_vals_shape[0])
+    {
+        throw py::value_error("Eigenvectors and eigenvalues have different shapes");
+    }
+
+    size_t src_nelems(1);
+
+    for (int i = 0; i < eig_vecs_nd; ++i)
+    {
+        src_nelems *= static_cast<size_t>(eig_vecs_shape[i]);
+    }
+
+    if (src_nelems == 0)
+    {
+        // nothing to do
+        return std::make_pair(sycl::event(), sycl::event());
+    }
+
+    // check compatibility of execution queue and allocation queue
+    if (!dpctl::utils::queues_are_compatible(exec_q, {eig_vecs, eig_vals}))
+    {
+        throw py::value_error("Execution queue is not compatible with allocation queues");
+    }
+
+    // check that arrays do not overlap, and concurrent access is safe.
+    // TODO: need to be exposed by DPCTL headers
+    // auto const &overlap = dpctl::tensor::overlap::MemoryOverlap();
+    // if (overlap(eig_vecs, eig_vals))
+    // {
+    //     throw py::value_error("Arrays index overlapping segments of memory");
+    // }
+
+    int eig_vecs_typenum = eig_vecs.get_typenum();
+    int eig_vals_typenum = eig_vals.get_typenum();
+    auto const& dpctl_capi = dpctl::detail::dpctl_capi::get();
+
+    sycl::event heevd_ev;
+    std::vector<sycl::event> host_task_events;
+
+    const std::int64_t n = eig_vecs_shape[0];
+    const oneapi::mkl::job jobz_val = static_cast<oneapi::mkl::job>(jobz);
+    const oneapi::mkl::uplo uplo_val = static_cast<oneapi::mkl::uplo>(upper_lower);
+
+    if ((eig_vecs_typenum == dpctl_capi.UAR_CDOUBLE_) && (eig_vals_typenum == dpctl_capi.UAR_DOUBLE_))
```

Comment:
You are assuming that both arrays are C-contiguous, I suppose, but the function does not check for that.

Reply:
`eig_vecs` is expected to be F-contiguous, while `eig_vals` are C-contiguous. I will add the check.